storage/gc_queue.go

```diff
@@ -239,10 +232,23 @@ func (gcq *gcQueue) process(now proto.Timestamp, repl *Replica) error {
 		}
 	}
 
+	done := true
 	if len(intents) > 0 {
+		done = false
 		repl.resolveIntents(repl.context(), intents)
 	}
 
+	// Set start and end keys.
+	if len(gcArgs.Keys) > 0 {
+		done = false
+		gcArgs.Key = gcArgs.Keys[0].Key
+		gcArgs.EndKey = gcArgs.Keys[len(gcArgs.Keys)-1].Key.Next()
+	}
+
+	if done {
+		return nil
+	}
+
 	// Send GC request through range.
```

Comment:
more specifically, when `len(keys) == 0`, nothing happens, `MVCCGarbageCollect` is really just a big range over those keys, with nothing before or after.